### PR TITLE
debug and trace level added to help set up the plugin.

### PR DESCRIPTION
**Pull request by kartikbha**
debug and trace level added to help set up the plugin. This will also help to trace who is log in to the system.

### CODE REVIEW

File: ckanext/ldap/controllers/user.py [hunk 1]
```diff
@@ -199,7 +199,18 @@ def _find_ldap_user(login):
...
     """
-    cnx = ldap.initialize(config['ckanext.ldap.uri'], bytes_mode=False)
+    
+    if config['ckanext.ldap.debug_level'] != '0':
+       ldap.set_option(ldap.OPT_DEBUG_LEVEL,config['ckanext.ldap.debug_level'])
+    else:
+       ldap.set_option(ldap.OPT_DEBUG_LEVEL,0)
+        
+    if config['ckanext.ldap.trace_level'] != '0':
+       cnx = ldap.initialize(config['ckanext.ldap.uri'], bytes_mode=False,config['ckanext.ldap.trace_level'])
```
Comment (jrdh) [inline ckanext/ldap/controllers/user.py]: This currently won't run as you have a positional argument after a keyword one (`SyntaxError: non-keyword arg after keyword arg`).

File: ckanext/ldap/plugin.py [hunk 2]
```diff
@@ -74,7 +74,9 @@ def configure(self, main_config):
             'ckanext.ldap.organization.role': {'default': 'member', 'validate': _allowed_roles},
             'ckanext.ldap.ckan_fallback': {'default': False, 'parse': p.toolkit.asbool},
             'ckanext.ldap.prevent_edits': {'default': False, 'parse': p.toolkit.asbool},
-            'ckanext.ldap.migrate': {'default': False, 'parse': p.toolkit.asbool}
+            'ckanext.ldap.migrate': {'default': False, 'parse': p.toolkit.asbool},
+            'ckanext.ldap.debug_level': {'default': '0'},
+            'ckanext.ldap.trace_level': {'default': '0'}
```
Comment (jrdh) [inline ckanext/ldap/plugin.py]: These could be defined as `{'default': 0, 'parse': p.toolkit.asint}` then you don't have to check the value in the code, you can just use it.

**Review by jrdh — CHANGES_REQUESTED** [review]
I've requested a couple of changes (see in line review comments), additionally, could you include the `trace_level` parameter in the [other initialize usage](https://github.com/NaturalHistoryMuseum/ckanext-ldap/blob/master/ckanext/ldap/controllers/user.py#L309)
 and then move the `ldap.set_option(ldap.OPT_DEBUG_LEVEL, ...)` code to the start of the file so that it always happens and only happens once?

**Comment by kartikbha** [comment]
issue in code so, closing

**Pull request closed without merging**